src/Backend/ONE/ONEToolchain.ts

```diff
@@ -14,33 +14,148 @@
  * limitations under the License.
  */
 
+import {execSync} from 'child_process';
+import * as vscode from 'vscode';
+
 import {Backend} from '../Backend';
 import {Command} from '../Command';
 import {Compiler} from '../Compiler';
 import {Executor} from '../Executor';
-import {Toolchains} from '../Toolchain';
+import {ToolchainInfo, Toolchains} from '../Toolchain';
+import {DebianArch, DebianRepo, DebianToolchain} from '../ToolchainImpl/DebianToolchain';
+import {Version} from '../Version';
+
+
+class ONEDebianToolchain extends DebianToolchain {
+  run(cfg: string): Command {
+    this.prepare();
+    let cmd = new Command('onecc-docker');
+    cmd.push('-C');
+    cmd.push(cfg);
+    return cmd;
+  }
+}
 
 class ToolchainCompiler implements Compiler {
   private readonly toolchainTypes: string[];
+  private readonly toolchainName: string;
+  private readonly debianRepo: DebianRepo;
+  private readonly debianArch: DebianArch;
 
   constructor() {
-    this.toolchainTypes = [];
+    this.toolchainName = 'onecc-docker-test';
+    // we currently support only the latest version of one-compiler.
+    this.toolchainTypes = ['latest'];
+
+    // we temporary add personal PPA for test.
+    this.debianRepo =
+        new DebianRepo('http://ppa.launchpad.net/psalmist-kim/ppa/ubuntu', 'bionic', 'main');
+    this.debianArch = DebianArch.amd64;
   }
 
   getToolchainTypes(): string[] {
     return this.toolchainTypes;
   }
 
-  getToolchains(_toolchainType: string, _start: number, _count: number): Toolchains {
-    throw Error('Invalid getToolchains call');
+  getVersion(version: string): Version {
+    let _version = version;
+    let option = '';
+
+    const optionIndex = version.search(/[~-]/);
+    if (optionIndex !== -1) {
+      option = version.slice(optionIndex);
+      _version = version.slice(0, optionIndex);
+    }
+
+    const [major = '0', minor = '0', patch = '0'] = _version.split('.');
+
+    return new Version(parseInt(major), parseInt(minor), parseInt(patch), option);
+  }
+
+  getToolchains(_toolchainType: string, _start?: number, _count?: number): Toolchains {
+    if (_toolchainType !== 'latest') {
+      throw Error(`Invalid toolchain type : ${_toolchainType}`);
+    }
+
+    let isToolchain: string;
+
+    try {
+      isToolchain =
+          execSync(`apt-cache show ${this.toolchainName} > /dev/null 2>&1; echo $?`).toString();
+    } catch (error) {
+      throw Error(`Getting ${this.toolchainName} package list is failed`);
+    }
+
+    if (parseInt(isToolchain.trim()) !== 0) {
+      throw Error('The prerequisites are not met');
+    }
+
+    const availableToolchains = new Toolchains();
+
+    let isToolchainVersion: string;
+    try {
+      isToolchainVersion =
+          execSync(`apt-cache show ${this.toolchainName} | grep Version | awk '{print $2}' | xargs`)
```

Comment:
We want to test `getToolchains` and `getInstalledToolchains`  and they have several `execSync` calls. We thought, they must be tested too. But we wonder whether it is the right way to use `try - catch` inside test code. And if so, we have to test catch case too. But how to make it throws an error in try statement is the question.